src/ledger/LedgerTxn.cpp

```diff
@@ -3620,8 +3663,10 @@ LedgerTxnRoot::Impl::getNewestVersion(InternalLedgerKey const& gkey) const
     }
 
     std::shared_ptr<LedgerEntry const> entry;
+
     try
     {
+        enforceBucketListPreLoadInvariants(key, mHeader->ledgerVersion);
```

Comment:
Shouldn't this line and the `enforceBucketListPostLoadInvariants` one below be inside the  `mApp.getConfig().isUsingBucketListDB()` conditional? It looks like the invariants will be checked in the SQL case as well unnecessarily.

Reply:
Ah yeah good catch, as-is it will be calling `enforceBucketListPostLoadInvariants` for the SQL case as well. I could move this call to inside the if, but IIRC I previously put it here with the intention of deleting the invariant enforcement in the SQL functions (for the sake of DRY), but never got around to it. Do you think its worth doing that? Or keep them separate and move this line inside the if.

Reply:
The invariants aren't 1 to 1 between SQL and buckets (ie the invariants in loadOffersByAccountAndAsset for example), so I think it makes sense to keep them separate. It's easier to reason about the invariants that way.

Reply:
That makes sense. I've updated it to enforce on the BL path only.